### PR TITLE
fix(bazel extractor): better handling of bazel exit code

### PR DESCRIPTION
**Pull request by justbuchanan**
We use the `--keep_going` flag so that we can extract all buildable targets even if some other targets fail.
If bazel fails to build some packages, but otherwise succeeds in extracting, the exit code is 1. Other
errors like "bazel server died unexpectedly" give other exit codes and will now cause the
extractor script to fail.

### CODE REVIEW

**Comment by justbuchanan** [comment]
@salguarnieri this should fix the issue where the tensorflow build OOMs 20% of the way through and reports a successful build.

**Comment by shahms** [comment]
It looks like the failure is related to bitbucket (I think @creachadair moved his repos to github).

**Comment by creachadair** [comment]
> It looks like the failure is related to bitbucket (I think @creachadair moved his repos to github).

I did, but not the ones Kythe depends on AFAIK.

**Comment by creachadair** [comment]
It is indeed complaining about https://godoc.org/bitbucket.org/creachadair/shell, but that package has not moved (and I wasn't planning to move it, specifically to avoid this kind of issue). Is it possible bitbucket had a transient issue? I've occasionally had their service hiccup on me.

**Comment by creachadair** [comment]
I wonder if maybe the bot hit a Bitbucket API rate limit?

```shell
$ curl https://api.bitbucket.org/2.0/repositories/creachadair/shell | jq
```
<details>

```
{
  "scm": "git",
  "website": "",
  "has_wiki": false,
  "name": "shell",
  "links": {
    "watchers": {
      "href": "https://api.bitbucket.org/2.0/repositories/creachadair/shell/watchers"
    },
    "branches": {
      "href": "https://api.bitbucket.org/2.0/repositories/creachadair/shell/refs/branches"
    },
    "tags": {
      "href": "https://api.bitbucket.org/2.0/repositories/creachadair/shell/refs/tags"
    },
    "commits": {
      "href": "https://api.bitbucket.org/2.0/repositories/creachadair/shell/commits"
    },
    "clone": [
      {
        "href": "https://bitbucket.org/creachadair/shell.git",
        "name": "https"
      },
      {
        "href": "git@bitbucket.org:creachadair/shell.git",
        "name": "ssh"
      }
    ],
    "self": {
      "href": "https://api.bitbucket.org/2.0/repositories/creachadair/shell"
    },
    "source": {
      "href": "https://api.bitbucket.org/2.0/repositories/creachadair/shell/src"
    },
    "html": {
      "href": "https://bitbucket.org/creachadair/shell"
    },
    "avatar": {
      "href": "https://bytebucket.org/ravatar/%7B9a6631f6-83b8-40dd-9bec-632d00b59290%7D?ts=go"
    },
    "hooks": {
      "href": "https://api.bitbucket.org/2.0/repositories/creachadair/shell/hooks"
    },
    "forks": {
      "href": "https://api.bitbucket.org/2.0/repositories/creachadair/shell/forks"
    },
    "downloads": {
      "href": "https://api.bitbucket.org/2.0/repositories/creachadair/shell/downloads"
    },
    "issues": {
      "href": "https://api.bitbucket.org/2.0/repositories/creachadair/shell/issues"
    },
    "pullrequests": {
      "href": "https://api.bitbucket.org/2.0/repositories/creachadair/shell/pullrequests"
    }
  },
  "fork_policy": "allow_forks",
  "uuid": "{9a6631f6-83b8-40dd-9bec-632d00b59290}",
  "language": "go",
  "created_on": "2015-12-29T22:56:26.753757+00:00",
  "mainbranch": {
    "type": "branch",
    "name": "master"
  },
  "full_name": "creachadair/shell",
  "has_issues": true,
  "owner": {
    "display_name": "Michael Fromberger",
    "uuid": "{a3fa73a0-b8a8-4be0-ba6b-d3df83248aed}",
    "links": {
      "self": {
        "href": "https://api.bitbucket.org/2.0/users/%7Ba3fa73a0-b8a8-4be0-ba6b-d3df83248aed%7D"
      },
      "html": {
        "href": "https://bitbucket.org/%7Ba3fa73a0-b8a8-4be0-ba6b-d3df83248aed%7D/"
      },
      "avatar": {
        "href": "https://avatar-management--avatars.us-west-2.prod.public.atl-paas.net/557058:a41d06e4-a248-4f95-abb8-e8828bcecf7c/dbd58213-b91c-4dcd-8d82-51b4640cab52/128"
      }
    },
    "nickname": "creachadair",
    "type": "user",
    "account_id": "557058:a41d06e4-a248-4f95-abb8-e8828bcecf7c"
  },
  "updated_on": "2019-06-03T16:36:35.994787+00:00",
  "size": 437958,
  "type": "repository",
  "slug": "shell",
  "is_private": false,
  "description": "Command-line splitting similar to the POSIX shell."
}
```
</details>

**Comment by shahms** [comment]
I manually triggered a rebuild which seemed to work.

**Comment by creachadair** [comment]
So maybe I _should_ move those last two packages. 😝

File: kythe/extractors/bazel/extract.sh [hunk 1]
```diff
@@ -66,6 +66,13 @@ fi
 # care if some targets fail, but bazel will return a failure code if any
 # targets fail.
 /kythe/bazelisk --bazelrc=/kythe/bazelrc "$@" || true
+retval=$?
+if [ $retval -eq 1 ]; then
+    echo "Not all bazel targets built successfully, but continuing anyways."
+else if [ $retval -ne 0 ];
```
Comment (shahms) [inline kythe/extractors/bazel/extract.sh]: You likely want to use the builtin `[[ ... ]]` or `(( ... ))` for these

Comment (justbuchanan) [reply kythe/extractors/bazel/extract.sh]: :+1:

File: kythe/extractors/bazel/extract.sh [hunk 2]
```diff
@@ -66,6 +66,13 @@ fi
 # care if some targets fail, but bazel will return a failure code if any
 # targets fail.
 /kythe/bazelisk --bazelrc=/kythe/bazelrc "$@" || true
+retval=$?
```
Comment (shahms) [inline kythe/extractors/bazel/extract.sh]: Have you tested this? I'm pretty sure this will always be zero due to the `|| true` unless pipefail is set.  You probably want: `${PIPESTATUS[0]}` to get the bazelisk status regardless of the rest of the pipeline or shell options.

Comment (shahms) [reply kythe/extractors/bazel/extract.sh]: Actually, that doesn't work either because it's not a pipeline, it's an `||`.  I think you'll need:
```
if !/kythe/bazelisk ...; then
  ...
fi
```

Comment (justbuchanan) [reply kythe/extractors/bazel/extract.sh]: Good catch, thanks. I changed from `|| true` to `|| retval=$?` and tested it this time.

File: kythe/extractors/bazel/BUILD [hunk 3]
```diff
@@ -11,7 +11,7 @@ docker_build(
         "@com_github_philwo_bazelisk//:bazelisk",
         "@javax_annotation_jsr250_api//jar",
     ],
-    image_name = "gcr.io/kythe-public/bazel-extractor",
```
Comment (shahms) [inline kythe/extractors/bazel/BUILD]: Was this change intentional?

Comment (justbuchanan) [reply kythe/extractors/bazel/BUILD]: Nope, fixed :)
